perun-core/src/main/java/cz/metacentrum/perun/core/impl/MembersManagerImpl.java

```diff
@@ -690,20 +691,62 @@ public MemberGroupStatus getUnifiedMemberGroupStatus(PerunSession sess, User use
 
 	@Override
 	public Paginated<Member> getMembersPage(PerunSession sess, Vo vo, MembersPageQuery query) {
-		return jdbc.query(
-				"SELECT " + memberMappingSelectQuery +
+
+		String userNameQueryString = Utils.prepareUserSearchQuerySimilarMatch();
+
+		Map<String, List<String>> attributesToSearchBy = Utils.getDividedAttributes();
+
+		Map<String, Pair<String, String>> attributesToSearchByQueries = Utils.getAttributesQuery(attributesToSearchBy.get("memberAttributes"), attributesToSearchBy.get("userAttributes"), attributesToSearchBy.get("uesAttributes"));
+
+		MapSqlParameterSource namedParams = Utils.getMapSqlParameterSourceToSearchUsersOrMembers(query.getSearchString(), attributesToSearchBy);
+		namedParams.addValue("voId", vo.getId());
+		namedParams.addValue("offset", query.getOffset());
+		namedParams.addValue("limit", query.getPageSize());
+
+		String statusesQueryString = "";
+		if (query.getStatuses() != null && !query.getStatuses().isEmpty()) {
+			statusesQueryString = " members.status in (:statuses) and ";
+			List<Integer> statusCodes = query.getStatuses().stream()
+				.map(Status::getCode)
+				.collect(Collectors.toList());
+			namedParams.addValue("statuses", statusCodes);
+		}
+
+		String idQueryString = "";
+		try {
+			int id = Integer.parseInt(query.getSearchString());
+			idQueryString = " members.user_id=" + id + " or members.id=" + id + " or ";
+		} catch (NumberFormatException e) {
+			// IGNORE wrong format of ID
+		}
+
+		Paginated<Member> members = namedParameterJdbcTemplate.query(
+				"SELECT DISTINCT " + memberMappingSelectQuery +
 				query.getSortColumn().getSqlSelect() +
 				", count(*) OVER() AS total_count" +
```

Comment:
With the current implementation, the count is not working, probably because of the added `DISTINCT` keyword.

Reply:
Thank you, it should be fixed now.